lib/prawn/images.rb

```diff
@@ -145,13 +144,14 @@ def verify_and_open_image(io_or_path)
         # read the file as binary so the size is calculated correctly
         # guard binmode because some objects acting io-like don't implement it
         io.binmode if io.respond_to?(:binmode)
-        return io
+        return io.read
       end
       # String or Pathname
       io_or_path = Pathname.new(io_or_path)
       raise ArgumentError, "#{io_or_path} not found" unless io_or_path.file?
-      io = io_or_path.open('rb')
-      io
+      File.open(io_or_path, 'rb') do |file_io|
```

Comment:
```ruby
File.read(io_or_path)
```
Instead of the block `open` would work as well.

Reply:
It would have to be `#binread` though, right?

Reply:
Yes, you're right. That is closer to the original code.